base/deprecated.jl

```diff
@@ -1093,6 +1093,24 @@ end
 #15995
 @deprecate symbol Symbol
 
+#15032: Expressions like Base.(:+) now call broadcast.  Since there is unlikely
+#       to be any legitimate reason to broadcast to a Module (not iterable!),
+#       it should be safe to catch this with a deprecation method.
+#       Note: this deprecation does not handle method definitions Base.(:+)(...) = ...,
+#             which are handled by the deprecate-dotparen function in julia-syntax.scm
+#       More generally, expressions like x.(sym) for any Symbol turn into broadcast
+#       but should really be getfield, and since you can't broadcast to a symbol
+#       using a deprecated broadcast method should be safe.   [Unfortunately,
+#       we can't catch x.(i::Int) in the same way, because broadcasting to an
```

Comment:
Actually, even if scalars aren't iterable, you still want `broadcast` to work for scalars.

* Things like `broadcast(atan2, 3, somearray)` are useful and need to work.  And it would be weird/annoying/non-generic if `broadcast` worked for mixed number/array arguments but not for pure-number arguments.
* It is pretty well-established that operations like `.+` should work for scalars as well as for vectors.  (Again, `3 .+ somearray` should work anyway, and having it work for scalars is important for writing generic code).
* By the same token, we probably want `f.(scalar)` to return a scalar `f(scalar)`.

So #7903 is orthogonal to this, I think.

Here is a compromise: `broadcast(f::Function, i::Integer)` calls the ordinary `broadcast`, while `broadcast(x, i::Integer)` otherwise gives a deprecation warning for `getfield`.